app/src/main/java/com/jerboa/feat/Voting.kt

```diff
@@ -28,10 +31,44 @@ data class InstantScores(
             score = upvotes - downvotes,
         )
     }
+
+    fun scoreOrPctStr(voteDisplayMode: VoteDisplayMode): String? {
+        return scoreOrPctStr(
+            score = score,
+            upvotes = upvotes,
+            downvotes = downvotes,
+            voteDisplayMode = voteDisplayMode,
+        )
+    }
 }
 
 // Set myVote to given action unless it was already set to that action, in which case we reset to 0
 fun newVote(
     oldVote: Int,
     voteAction: VoteType,
 ): Int = if (voteAction.value == oldVote) 0 else voteAction.value
+
+fun upvotePercent(
+    upvotes: Long,
+    downvotes: Long,
+): Float {
+    return (upvotes.toFloat() / (upvotes + downvotes))
+}
+
+fun formatPercent(pct: Float): String {
+    val formatted = "%.0f".format(pct * 100F)
+    return "$formatted%"
```

Comment:
I think you can do this in one line by just escaping the `%`

Reply:
That ended up not working `.format(` needs only formatting characters, but I made this more compact now.

Reply:
Ah yeah that did. Originally I tried `\\%` which failed.